src/test/java/com/ordestiny/tdd/designpattern/factorypattern/AnimalFactoryTest.java

```diff
@@ -0,0 +1,77 @@
+package com.ordestiny.tdd.designpattern.factorypattern;
+
+import com.ordestiny.tdd.designpattern.factorypattern.animal.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * @author Rachit Parikh
+ */
+
+public class AnimalFactoryTest {
+
+    Animal animal;

```

Comment:
What is the purpose of this global variable?

Reply:
I created it because this single object was used by each and every test class. So instead of creating local variables multiple times, I created a single global variable which will be overwritten when each test class is run. It was just to avoid complexity.